### PR TITLE
ci: test ONNX release candidates weekly

### PR DESCRIPTION
**Pull request by neallin917**
## Summary

- add a weekly and manually dispatchable ONNX release-candidate compatibility workflow
- select only non-yanked prereleases newer than the latest stable PyPI release
- skip cleanly when no active ONNX release candidate exists
- reuse the existing unit-test composite action and upload its JUnit results
- add focused tests for prerelease selection, including stale, yanked, dev, invalid, and empty releases

## Why

The release-integration roadmap in #2472 calls for a recurring ONNX RC lane so compatibility problems are found before each stable ONNX release instead of through one-off integration issues. The existing matrix covers fixed ONNX versions but cannot discover future release candidates automatically.

The workflow downloads official PyPI metadata with retries, resolves the current release candidate using PEP 440 ordering, and runs the established TensorFlow 2.15 / Python 3.11 compatibility suite at opset 18. When PyPI has no prerelease newer than stable, the run records that state in the job summary and exits successfully.

Addresses the recurring ONNX-RC integration follow-up in #2472.

## Validation

- `python -m unittest tools/test_latest_pypi_prerelease.py` on Python 3.11
- real PyPI metadata smoke test (currently resolves no active RC, as expected)
- `ruff check tools/latest_pypi_prerelease.py tools/test_latest_pypi_prerelease.py`
- `actionlint .github/workflows/onnx_release_candidate_ci.yml`
- YAML parse and `git diff --check`

Signed-off-by: Neal <neal.lin917@gmail.com>

### CODE REVIEW

File: tools/test_latest_pypi_prerelease.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            "1.21.0": [{"yanked": False}],
+            "1.22.0rc1": [{"yanked": False}],
+            "1.22.0": [{"yanked": False}],
+        }
+
+        self.assertEqual(latest_prerelease(releases), "")
+
+
+if __name__ == "__main__":
+    unittest.main()
```
[the remark this answers is not in the document]
Comment (neallin917) [reply tools/test_latest_pypi_prerelease.py:37]: Addressed in e274c5c7. Added `test_returns_empty_when_no_releases_exist`, and `python3 -m unittest tools/test_latest_pypi_prerelease.py` now passes all 3 tests.

File: .github/workflows/onnx_release_candidate_ci.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test-onnx-release-candidate:
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@9c091bb21b7c1c1d1991bb908d89e4e9dddfe3e0 # v7.0.0
```
[the remark this answers is not in the document]
Comment (neallin917) [reply .github/workflows/onnx_release_candidate_ci.yml]: Addressed in e274c5c7 after rebasing onto the updated PR branch. The workflow now uses the repository's v7.0.1 checkout pin (`3d3c42e5...`), matching the current main workflows.
